### DISCUSSION TITLE
Which is "official" form lib for PrimeVue? Is it still VeeValidate?

### DISCUSSION THREAD

**Discussion by liborjelinek** [primefaces/community#2022] [PrimeVue]
A some time ago it was [VeeValidate](https://vee-validate.logaretm.com/v4/). They still has "Works with PrimeVue" on their homepage, but link is now invalid and PrimeVue 3.52.0 docs doesn't contain information on integration anymore...

We're greenfield project. Which form lib do you recommend (which planned changes, like more "Tailwindish", in mind)? VeeValidate, FormKit or rather not to say?

**Comment by 23457890** [comment]
Would be great to get an answer on this. The current options do not seem great.

I believe I saw a post from the founder on Reddit about PrimeVue eventually adding form validation, but I could be mistaken and do not see it on the roadmap.

Having some type of form context to work with and the ability to use a validation library like Yup would be ideal.

**Comment by mateenagy** [comment]
I also saw the [reddit post ](https://www.reddit.com/r/vuejs/comments/1d2lthp/comment/l61dxl9/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)where they said they will provide built-in form library in v4.1 but it will be available in the Q3 this year (hopefully).

In the meantime I can suggest to take a look in [VueFormify](https://vue-formify.matenagy.me/). I implement `PrimeVue` form components so it can be implemented with ease. You can use `yup/zod/joi/valibot` for validation.

**Comment by liborjelinek** [comment]
Mmm, overaly, it's very unstable topic with PrimeVue. Veevalidate was "removed" with no mention even within 3.x versions.

After a long discussion, our team, has decided to go with https://vueform.com. It's complete solution with validation and rendering on its own with optional commercial support. Default theme looks like PrimeVue theme, so nobody notice any visual inconsistence.

**Reply by zanyar3** [reply]
> Mmm, overaly, it's very unstable topic with PrimeVue. Veevalidate was "removed" with no mention even within 3.x versions.
> 
> After a long discussion, our team, has decided to go with https://vueform.com. It's complete solution with validation and rendering on its own with optional commercial support. Default theme looks like PrimeVue theme, so nobody notice any visual inconsistence.

Can you use **primevue** components instead **vueform** components ? 
Is any documentation available for the integration primevue with vueform ?

**Reply by liborjelinek** [reply]
> > Mmm, overaly, it's very unstable topic with PrimeVue. Veevalidate was "removed" with no mention even within 3.x versions.
> > After a long discussion, our team, has decided to go with https://vueform.com. It's complete solution with validation and rendering on its own with optional commercial support. Default theme looks like PrimeVue theme, so nobody notice any visual inconsistence.
> 
> Can you use **primevue** components instead **vueform** components ? Is any documentation available for the integration primevue with vueform ?

Don't know if and how. In my case, I am very happy with Vueform and see no reason to embed PrimeVue components into Vueform. Mainly because default PrimeVue and Vueform looks almost the same. User's don't notice the difference. But Vueform is generally well documented and I expect it would be possible.

**Reply by 23457890** [reply]
After this question was asked, it was confirmed PrimeVue is adding it's own form validation logic.  See Q3 in the roadmap (I haven't been following closely since then, so not sure of the status).

https://primevue.org/roadmap/

**Reply by j0rgedev** [reply]
@23457890 form validation with be available in primevue 4.2.0
